### PR TITLE
Update release target in Makefile.

### PR DESCRIPTION
**Pull request by mikecanoy**
Also updated `dist` and `install` targets from `python` to `python3` since `setup.py` has type hints.
```python
def get_version(string: str) -> str:
```

### CODE REVIEW

**Comment by reillysiemens** [comment]
@solus-impar: Wow! Thanks! At first glance this looks good, but I'm going to quick read through the PyPA [Python Packaging User Guide](https://packaging.python.org/tutorials/distributing-packages/) before merging.

File: Makefile [hunk 1]
```diff
@@ -72,12 +72,12 @@ tag-release: check-origin-remote
 	git tag -a $(VERSION) -m "Release $(VERSION)"
 	git push origin $(VERSION)
 
-release: tag-release ## package and upload for public release
-	# TODO: Use twine maybe?
-	@echo Honestly, what should this do?
+release: dist ## package and upload for public release
```
Comment (reillysiemens) [inline Makefile]: This is almost what I want, but I think I'd still like to have the `tag-release` target in here. I'd like to be able to run `make release` and have signing, upload to PyPI, and tagging the commit/pushing that to the repo all in one fell swoop.

**Review by reillysiemens — CHANGES_REQUESTED** [review]
This is exactly what the packaging guide [recommends](https://packaging.python.org/tutorials/distributing-packages/#upload-your-distributions) so this will work for me. Address my one concern and I'll be happy to merge this in. Thanks again!

File: Makefile (hunk 1 above)
Comment (reillysiemens) [reply Makefile]: The `dist` target should probably come before `tag-release`.

Comment (mikecanoy) [reply Makefile]: I couldn't verify it, but I think `relase: tag-release dist` would work. Does `make tag-release` hag during `sed` for you?

Comment (reillysiemens) [reply Makefile]: Hmm... it _shouldn't_ hang during `sed`, but it's possible I made an error in that gnarly expression. Let me double check.

Comment (reillysiemens) [reply Makefile]: Nah, pulling the `sed` expression out on its own yields the correct result:
```bash
(layabout) [23:28:15] tucker@khanti:~/Projects/layabout git:(implement-layabout) $ sed -rn "s/__version__ = [\'\"](.*)[\'\"]/\1/p" layabout.py 
1.0.0b
```

Comment (reillysiemens) [reply Makefile]: Heh. I just tried to run this myself though and it's definitely stuck somewhere. I think it's stuck after executing `check-origin-remote`, but before it really gets to `tag-release`...

Comment (reillysiemens) [reply Makefile]: Oh, no, you were right. It's the fact that `$(VERSION_FILE)` is undefined, so `sed` is probably expecting input from stdin. If you swap out `$(VERSION_FILE)` for `$(SRC)` you'll get the right result. This is a classic copy pasta error on my part.

File: Makefile [hunk 2]
```diff
@@ -66,18 +66,18 @@ check-origin-remote:
 	# A remote named origin MUST exist.
 	@git remote show origin
 
+dist: clean ## build source and wheel packages
```
Comment (reillysiemens) [inline Makefile]: Oh! I'm really sorry, I didn't mean to say that the `dist` target should come before the `tag-release` target in general, just in the target list for `release` like
```make
release: dist tag-release
    # ...
```
That way I think I'd avoid accidentally tagging a release if an error occurred in building a dist.

If you wouldn't mind moving the `dist` target back to where it was originally then I'll really be happy to merge this. Promise! 😅

**Review by reillysiemens — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (reillysiemens) [reply Makefile]: If you could also squash these and any subsequent commits that would be great. 😁

**Review by reillysiemens — APPROVED** [review]
[LGTM](https://www.lgtm.in/i/Ao7Nuc6hq)

**Comment by mikecanoy** [comment]
:tada:

**Pull request merged**